lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -940,67 +941,85 @@ def alerts(self) -> dict:
                 continue
 
             alert_rules = json.loads(relation.data[relation.app].get("alert_rules", "{}"))
-
             if not alert_rules:
                 continue
 
-            if "groups" not in alert_rules:
-                logger.debug("No alert groups were found in relation data")
+            identifier, topology = self._get_identifier_by_alert_rules(alert_rules)
+            if not topology:
+                try:
+                    scrape_metadata = json.loads(relation.data[relation.app]["scrape_metadata"])
+                    identifier = JujuTopology.from_dict(scrape_metadata).identifier
+                    alerts[identifier] = self._tool.apply_label_matchers(alert_rules)  # type: ignore
+
+                except KeyError as e:
+                    logger.debug(
+                        "Relation %s has no 'scrape_metadata': %s",
+                        relation.id,
+                        e,
+                    )
```

Comment:
Is there a `continue` missing here?

Reply:
No, there isn't.

The `continue` is effectively the next line, where `if not identifier:` evalues to "if `self._get_identifier...` didn't return a `topology`, it may have returned a bare string as `identifier`. If there's no `topology`, and we can't build an `identifier` from `JujuTopology.from_dict(...)`", the bare string is the fallthrough.

At that point, `if not identifier` is checking the failure case of both the `try:` block and `_get_identifier_by_...`